libathemecore/services.c

```diff
@@ -617,10 +617,20 @@ handle_certfp(struct sourceinfo *si, struct user *u, const char *certfp)
 	if (u->myuser != NULL)
 		return;
 
-	if ((mcfp = mycertfp_find(certfp)) == NULL)
+	if ((mcfp = mycertfp_find(certfp)) != NULL)
+		mu = req.mu = mcfp->mu;
+
+	req.si = si;
```

Comment:
This needs to initialise `req.mu` with a value, or its value will be undefined in the hook functions that access it.

You can do this by initialising `mu` to `NULL` at the top of this function, and then adding `req.mu = mu;` here, taking the `req.mu` assignment out of the conditional block above.